src/Microsoft.VisualStudio.LanguageServices.Razor/DocumentStructureChangedEventArgs.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.VisualStudio.LanguageServices.Razor
+{
+    public sealed class DocumentStructureChangedEventArgs : EventArgs
+    {
+        public DocumentStructureChangedEventArgs(
+            SourceChange change,
+            ITextSnapshot buffer,
+            RazorCodeDocument codeDocument)
+        {
+            SourceChange = change;
+            Buffer = buffer;
+            CodeDocument = codeDocument;
+        }
+
+        /// <summary>
+        /// The <see cref="AspNetCore.Razor.Language.SourceChange"/> which triggered the re-parse.
+        /// </summary>
+        public SourceChange SourceChange { get; }
```

Comment:
Should be `SourceChange` 😄. They don't go between source changes and textchanges shockingly.